src/pngwriter.h

```diff
@@ -196,6 +200,8 @@ class pngwriter
     * */
    pngwriter & operator = (const pngwriter & rhs);
 
+   friend void swap(pngwriter & lhs, pngwriter & rhs);
```

Comment:
can you pls add (doxygen) header comments?

Why don't you implements this as a private member?